Allow filtering by field names in GET resources for REST Data with Panache

Sgitario opened the pull request:
With these changes, the generated GET resource by REST DATA with Panache to list all the entities accepts filtering by the entity fields. For example, calling `http://localhost:8080/people?name=Peter Peterson` would return:

```json
[
  {
    "id": 2,
    "name": "Peter Peterson",
    "birth": "1986-11-20"
  }
]
```

Sgitario commented:
PR updated with fixing failures for MongoDB (it seems that panache is not behaving the same for MongoDB when providing empty queries). 
Also, instead of getting the query params from the URI info, now we're getting the query params directly from the method using the annotation `@QueryParam`. This approach has many advantages:
- RESTEasy / RESTEasy Reactive will convert the values from string to the correct field type (but on purpose only a few are supported).
- It's already integrated with the OpenAPI schema.

geoand commented on docs/src/main/asciidoc/rest-data-panache.adoc:
```diff
@@ -354,6 +354,39 @@ It applies to the paged resources only and is a number starting with 1. Default
+]
+----
+
+IMPORTANT: Filtering by fields only supports for primitive types.
```
```suggestion
IMPORTANT: Filtering by fields is only supported for primitive types.
```

geoand commented:
Please squash the commits :)

Sgitario committed:
Allow filtering by field names in GET resources for REST Data with Panache

With these changes, the generated GET resource by REST DATA with Panache to list all the entities accepts filtering by the entity fields. For example, calling `http://localhost:8080/people?name=Peter Peterson` would return:

```json
[
  {
    "id": 2,
    "name": "Peter Peterson",
    "birth": "1986-11-20"
  }
]
```

Sgitario commented:
> Please squash the commits :)

Done

rsvoboda commented:
I think this PR could be labeled as noteworthy or/and breaking change. I think there should be a note in https://github.com/quarkusio/quarkus/wiki/Migration-Guide-2.15

We have noticed failures in our TS, Antonio was discussing this with Pablo.

We were overriding the `findAll` method from the PanacheRepositoryBase class, and the change introduced here is no longer calling that `findAll` method internally but the `find` method.

So if users/customers are currently implementing findAll to override the current default list behaviour, it won't work after migrating to the 2.15 version.

We can speculate if overriding is something that users should do or not. But they technically can and I bet they will do that if that fixes their trouble. Thus I think we should notify them in Migration-Guide-2.15 and it can be also used as an opportunity to highlight that overrides of generated stuff can be risky in the long run as things can change and evolve.

Sgitario commented:
> I think this PR could be labeled as noteworthy or/and breaking change. I think there should be a note in https://github.com/quarkusio/quarkus/wiki/Migration-Guide-2.15
> 

These changes will be listed in the noteworthy release notes along with https://github.com/quarkusio/quarkus/pull/29323 (we labeled the pull request #29323 instead of this pull request). 

> We were overriding the findAll method from the PanacheRepositoryBase class, and the change introduced here is no longer calling that findAll method internally but the find method.

To give more context, before these changes, the auto-generated resources were calling to the `PanacheRepositoryBase.findAll()` and `PanacheRepositoryBase.findAll(sort)` methods. And after these changes, the auto-generated resources are now calling the `PanacheRepositoryBase.find(query, params)` and `PanacheRepositoryBase.find(query, sort, params)` methods. This is totally transparent for users, so they don't need to migrate anything... Unless you override either the `findAll()` or the `findAll(sort)` of the entity repository because now these methods won't be called by the auto-generated resources.  

So, for me, this is a bit cumbersome to note in the migration guide (as users won't likely need to migrate anything). However, let's see what @geoand thinks about this and perhaps we can add a note saying that this only affects users overriding the mentioned repository methods.
